### DISCUSSION TITLE
How to get certificates of remote host?

### DISCUSSION THREAD

**Discussion by stephenmw** [strongswan/strongswan#1581] [Q&A]
I have a remote ipsec server for which I do not have the CA certificates or server certificate. Is there a way to capture that information through strongswan so I can add it to the configuration? Once I have the issuer certificate, I can add it to /etc/ipsec.d/cacerts.

**Comment by tobiasbrunner** [comment]
The (root) CA certificate is never transmitted, so you have to get that directly from the server admin (or if the server cert is from a third-party CA get it from there or your distro, check the log for the subject DN of the certificate's issuer).

The server certificate might be transmitted (depends on the server config), but there is no easy way to get at it for failed connections (i.e. you'd have to use a high enough log level and extract the hex dump of the certificate from that). And you'd then also need some other way to verify it's the correct certificate so you could potentially trust it.
